### PR TITLE
VACMS-16061: updated graphql to v3.4

### PR DESCRIPTION
**Pull request by edmund-dunn**
## Description

Relates to #16061

## Testing done
Tested locally by running content build, build completed successfully. 
```
Finished 918 queries in 2032s with 27284 pages
https://va-gov-cms.ddev.site response time: : 33:53.878 (m:ss.mmm)
Total time to load content from GraphQL: 34:10.405 (m:ss.mmm)
Drupal successfully loaded!
Successfully piped Drupal content into Metalsmith!
```

## Screenshots
CleanShot2023-11-08at14 00 41
CleanShot2023-11-08at14 09 55

## QA steps

What needs to be checked to prove this works?
What needs to be checked to prove it didn't break any related things?
What variations of circumstances (users, actions, values) need to be checked?

1. View the web front-end tugboat preview https://web-39f99ty0oncifd1xdnvs9v8m8hyaeerm.ci.cms.va.gov/
   - [ ] Validate that it is built

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [x] `CMS Team`
- [x] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [x] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by eselkin — APPROVED** [review]
lgtm

File: composer.json [hunk 1]
```diff
@@ -426,10 +426,6 @@
             "drupal/flysystem_s3": {
                 "3297257 - Automated Drupal 10 compatibility fixes": "https://www.drupal.org/files/issues/2023-04-27/3297257-drupal-10-compatibility-fixes_0.patch"
             },
-            "drupal/graphql": {
-                "Fix Explorer with Layout Builder enabled - https://github.com/drupal-graphql/graphql/pull/1144": "patches/graphql_8x3x_layout_builder_incompat.patch",
-                "Remove outdated RouteQueryEnhancer controller reference - https://github.com/drupal-graphql/graphql/pull/1230": "patches/graphql_8x3x_query_route_enhancer_update.patch"
-            },
```
Comment (swirtSJW) [inline composer.json:432]: @edmund-dunn there are two patches being removed.  Is it confirmed that those two patches are now part of v3.4?

**Comment by swirtSJW** [comment]
Release notes
- [3.4](https://www.drupal.org/project/graphql/releases/8.x-3.4)
   - [GraphQL - Moderately critical - Access bypass - SA-CONTRIB-2023-050](https://www.drupal.org/sa-contrib-2023-050)
   - [GraphQL - Moderately critical - Cross Site Request Forgery - SA-CONTRIB-2023-051](https://www.drupal.org/sa-contrib-2023-051)
- [3.3](https://www.drupal.org/project/graphql/releases/8.x-3.3)
   - MAJOR UPDATE - Drupal 10 compatibility for 3.x
   - The "graphql-php" library has been updated to one of the latest versions, making it compatible with PHP8. This can cause compatibility breaks with custom code within existing projects.
   - This version is no longer compatible with Drupal 8.
   - The code has been updated to stop using deprecated functions in Drupal 10 and also to make it compatible with PHP8.
   - The graphql explorer and voyager libraries have also been updated with the same changes as in the 4.x branch. This way, the core library "core/once" is used, and "jquery.once" is no longer used.
   - We recommend upgrading from 3.x to 4.x to take advantage of the improvements made in the GraphQL module and the underlying GraphQL library. If you still require automatic generation of the schema then take a look at the GraphQL Compose or GraphQL Core Schema projects which have implemented automatic schema generation on top of the 4.x GraphQL module.
-  3.2
   - MAJOR UPDATE - Drupal 10 compatibility for 3.x
   - The "graphql-php" library has been updated to one of the latest versions, making it compatible with PHP8. This can cause compatibility breaks with custom code within existing projects.
   - This version is no longer compatible with Drupal 8.
   - The code has been updated to stop using deprecated functions in Drupal 10 and also to make it compatible with PHP8.
   - The graphql explorer and voyager libraries have also been updated with the same changes as in the 4.x branch. This way, the core library "core/once" is used, and "jquery.once" is no longer used.
   - We recommend upgrading from 3.x to 4.x to take advantage of the improvements made in the GraphQL module and the underlying GraphQL library. If you still require automatic generation of the schema then take a look at the GraphQL Compose or GraphQL Core Schema projects which have implemented automatic schema generation on top of the 4.x GraphQL module.

File: composer.json (hunk 1 above)
Comment (swirtSJW) [reply composer.json:432]: Looks like the first one is 
image

Comment (swirtSJW) [reply composer.json:432]: And the second one is good too.

Comment (edmund-dunn) [reply composer.json:432]: @swirtSJW yes, I don't remove patches without doing that.

**Comment by swirtSJW** [comment]
- [x] Checked revisions logs for breaking changes
- [x] Checked removed patches to make sure they made it into this version
- [x] https://pr16070-39f99ty0oncifd1xdnvs9v8m8hyaeerm.ci.cms.va.gov/graphql/explorer  Checked explorer to be operational
- [ ] Check Voyager https://pr16070-39f99ty0oncifd1xdnvs9v8m8hyaeerm.ci.cms.va.gov/graphql/voyager

**Comment by swirtSJW** [comment]
Voyager seems to be non operational when compared to prod where it is working 
PR on left, prod on right.
image

**Comment by swirtSJW** [comment]
Also seeing an error coming out of a test
https://pr16070-39f99ty0oncifd1xdnvs9v8m8hyaeerm.ci.cms.va.gov/admin/reports/dblog?type%5B%5D=graphql
image

**Review by swirtSJW — CHANGES_REQUESTED** [review]
I am seeing two defects that need to be checked out
- [ ] Voyager not working
- [ ] Error coming from test (need to check other pr environments to see if they also include that error to determine if it is coming from this update or not
image

**Comment by edmund-dunn** [comment]
> Also seeing an error coming out of a test https://pr16070-39f99ty0oncifd1xdnvs9v8m8hyaeerm.ci.cms.va.gov/admin/reports/dblog?type%5B%5D=graphql image

This is an existing error. I see it on staging.
CleanShot2023-11-09at07 43 59

**Comment by edmund-dunn** [comment]
> Voyager seems to be non operational when compared to prod where it is working PR on left, prod on right. image

Voyager is not functioning correctly on prod either. If Voyager is needed then an issue can be entered and we will get it working. Since this has never been brought up before I think we can safely assume that its notbeing used by anyone at this time. @ndouglas thoughts?
CleanShot2023-11-09at08 08 50

**Comment by ndouglas** [comment]
Voyager's never worked. 

We're not going to delay a security update because someone we don't really use and has never worked correctly seems to be broken in a different way than previously.

If someone wants Voyager to work, then they can open an issue for it.

**Comment by ndouglas** [comment]
According to [this ticket](https://github.com/drupal-graphql/graphql/issues/1238), Voyager might not have worked for a long time on 3.x

**Comment by swirtSJW** [comment]
Voyager is throwing an error on prod and is not drawing the graph, but it still allows a FE developer to explore the available fields on a content type 
image


That can not be done at all after this upgrade.
image

It does not need to hold up the security update (which zero trust or not is a non-issue because it is only preventing someone from reading our data, which is all publicly available anyway. Passwords would be the only thing at risk... if graphQL could surface them, which it can not.)   But I do not want to bury this as, it has always been broken when it is now more broken than it has been.

Also there is some history of other teams asking for it in the past
- https://dsva.slack.com/archives/C0FQSS30V/p1604585083323700?thread_ts=1604584812.323400&cid=C0FQSS30V  Michael Pelz Sherman in a conversation with Dave Conlon
- https://dsva.slack.com/archives/CDHBKAL9W/p1641741179016900 Napoleon Kerzant in a conversation with Ryan and I.

I swear Neil got it working after Napoleon requested it, but I can find no record of it now.

**Review by swirtSJW — APPROVED** [review]
My concerns were addressed by feedback from @ndouglas and @edmund-dunn so I am approving this.

**Review by dsasser — DISMISSED** [review]
LGTM

**Comment by ndouglas** [comment]
All I did was resolve a merge conflict, so I'll merge once tests pass.

**Pull request merged**